src/ext/JourneyPatternStopPointMap/QuayMarker.tsx

```diff
@@ -0,0 +1,122 @@
+import { Quay, StopPlace } from '../../api';
+import { getMarkerIcon } from './markers';
+import { Marker, Popup } from 'react-leaflet';
+import { Button } from '@entur/button';
+import { useIntl } from 'react-intl';
+import { Heading5 } from '@entur/typography';
+
+interface QuayMarkerProps {
+  quay: Quay;
+  stopPlace: StopPlace;
+  stopPointIndex: number | undefined;
+  hasSelectedQuay: boolean;
+  hasNonSelectedQuays: boolean;
+  hideNonSelectedQuaysState: boolean;
+  hideNonSelectedQuaysCallback: (hideNonSelected: boolean) => void;
+  showQuaysCallback: (showAll: boolean) => void;
+  addStopPointCallback: (quayId: string) => void;
+  deleteStopPointCallback: (index: number) => void;
+}
+
+const QuayMarker = ({
+  quay,
+  stopPointIndex,
+  stopPlace,
+  hasSelectedQuay,
+  hasNonSelectedQuays,
+  showQuaysCallback,
+  hideNonSelectedQuaysCallback,
+  hideNonSelectedQuaysState,
+  addStopPointCallback,
+  deleteStopPointCallback,
+}: QuayMarkerProps) => {
+  const intl = useIntl();
+  const { formatMessage } = intl;
+  return (
+    <Marker
+      key={'quay-marker-' + quay.id}
+      icon={getMarkerIcon(
+        stopPlace.transportMode,
+        false,
+        stopPointIndex !== undefined,
+      )}
+      position={[
+        quay.centroid?.location.latitude,
+        quay.centroid?.location.longitude,
+      ]}
+    >
+      <Popup>
+        <section>
+          <Heading5 className={'popup-title'}>{stopPlace.name.value}</Heading5>
+          <div className={'popup-id'}>{quay.id}</div>
+        </section>
+
+        <section>
+          {formatMessage({ id: stopPlace.transportMode?.toLowerCase() })}
+        </section>
+
+        <div className={'popup-button-panel'}>
+          <Button
+            className={'popup-button'}
+            style={{
+              marginRight: '0.5rem',
+            }}
+            onClick={() => {
+              // If the quay got unselected and hideNonSelectedQuaysState mode was on, then switching to a stop place view:
+              if (stopPointIndex !== undefined && hideNonSelectedQuaysState) {
+                showQuaysCallback(false);
+                hideNonSelectedQuaysCallback(false);
+              }
+
+              // Callback to update the GenericStopPointsEditor form state:
+              if (stopPointIndex !== undefined) {
```

Comment:
This area gets refactored in the next PR about supporting multiple stop point indexes